test_scripts/iAP2TransportSwitch/002_resume_failed_simple_case.lua

```diff
@@ -0,0 +1,164 @@
+--  SDL, HMI are running on system.
+--
+--  2. Performed steps
+--  iAP2 Bluetooth mobile device connects to system
+--  appID_1->RegisterAppInterface(params)
+--  RAI response is SUCCESS
+--  application sends SubscribeButton for PRESET_0 button (resume data)
+--  SDL sends OnHashChange to application
+--
+--  same iAP2 mobile device is connected over USB to system and re-registers within AppTransportChangeTimer timeout
+--  appID_1->RegisterAppInterface(params), but with wrong hashID
+--
+--  Expected behavior:
+--  1. SDL successfully registers application and notifies HMI and mobile
+--     SDL->HMI: OnAppRegistered(params)
+--     SDL->appID: SUCCESS, success:"true":RegisterAppInterface()
+--     SDL->appID: SUCCESS, success:"true": SubscribeButtons()
+--     SDL->appID: OnHashChange
+--
+--  2. SDL successfully registers application and notifies mobile only with RAI response
+--     SDL->appID: RESUME_FAILED, success:"true":RegisterAppInterface()
+--     application remains registered internally
+--     application does not send OnAppUnregistered notification to HMI
+--     application does not send OnAppRegistered notification to HMI
+--     SDL->HMI: Buttons.OnButtonSubscription, isSubscribed = false, name = PRESET_0
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/iAP2TransportSwitch/common')
+local mobSession = require("mobile_session")
+
+--[[ Local Variables ]]
+local deviceBluetooth
+local sessionBluetooth
+
+--[[ Local Functions ]]
+local function connectBluetoothDevice(self)
+  deviceBluetooth = self:createIAP2Device(common.device.bluetooth.id,
+    common.device.bluetooth.port, common.device.bluetooth.out)
+
+  EXPECT_HMICALL("BasicCommunication.UpdateDeviceList", {
+    deviceList = {
+      {
+        id = config.deviceMAC,
+        name = common.device.bluetooth.uid,
+        transportType = common.device.bluetooth.type
+      }
+    }
+  })
+  :Do(function(_, data)
+      self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", { })
+    end)
+
+  self:connectMobile(deviceBluetooth)
+  :Do(function()
+      sessionBluetooth = mobSession.MobileSession(self, deviceBluetooth, common.appParams)
+      sessionBluetooth:StartService(7)
+      :Do(function()
+          local cid = sessionBluetooth:SendRPC("RegisterAppInterface", common.appParams)
+          sessionBluetooth:ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+        end)
+      EXPECT_HMICALL("BasicCommunication.UpdateAppList")
+      :Do(function(_, data)
+          self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", { })
+        end)
+      EXPECT_HMINOTIFICATION("Buttons.OnButtonSubscription", {
+        isSubscribed = true,
+        name = "CUSTOM_BUTTON"
+      })
```

Comment:
Why is this subscription expected? I don't see any corresponding `SubscribeButton` RPC sent from the mobile session

Reply:
@jacobkeeler This event is expected due to SDL requirement:

> By default, SDL must subscribe every application to CUSTOM_BUTTON right after the application's registration ONLY if system supports CUSTOM_BUTTON